Products/ZenHub/metricpublisher/tests/test_publisher.py

```diff
@@ -0,0 +1,441 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2014, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+from unittest import TestCase, TestSuite, makeSuite
+from mock import Mock, MagicMock, create_autospec, patch
+from zope.interface.verify import verifyObject
+
+import ujson as json
+
+from Products.ZenHub.metricpublisher.publisher import (
+    os,
+    sys,
+    defer,
+    deque,
+    RedisListPublisher,
+    HttpPostPublisher,
+    BasePublisher,
+    StringProducer,
+    ResponseReceiver,
+    IBodyProducer,
+    RedisClientFactory,
+    Headers,
+    CookieJar,
+    CookieAgent,
+    UNAUTHORIZED,
+    INITIAL_REDIS_BATCH,
+    defaultPublishFrequency,
+    defaultMetricBufferSize,
+    basic_auth_string_content
+)
+
+
+METRIC_BUILDED = [
+    '{"timestamp":1535460634.26479,"metric":"eventQueueLength",\
+    "value":0.0,"tags":{\
+        "instance":"0","daemon":"zenpython","monitor":"localhost",\
+        "metricType":"GAUGE","tenantId":"b0e4t72hrole5z1xv88djc1hb"\
+        }\
+    }'
+]
+METRIC = 'testMetricName'
```

Comment:
BUFFER_LEN is only used in BasePublisherTest, and should be defined within its scope until it is needed by another class.

Reply:
I also use BUFFER_LEN  in RedisPublisherTest.test_init